endpoint_info.go

```diff
@@ -211,9 +211,6 @@ func (ep *endpoint) AddStaticRoute(destination *net.IPNet, routeType int, nextHo
 }
 
 func (ep *endpoint) addInterfaceRoute(route *types.StaticRoute) error {
-	ep.Lock()
-	defer ep.Unlock()
-
```

Comment:
+1. I think it causes unnecessary confusion over when to lock and when not to lock